src/test/fuzz/kitchen_sink.cpp

```diff
@@ -2,15 +2,27 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
+#include <addrdb.h>
+#include <chainparams.h>
+#include <net.h>
+#include <outputtype.h>
+#include <policy/fees.h>
 #include <rpc/util.h>
 #include <test/fuzz/FuzzedDataProvider.h>
 #include <test/fuzz/fuzz.h>
 #include <test/fuzz/util.h>
+#include <test/util/setup_common.h>
 #include <util/error.h>
 
 #include <cstdint>
 #include <vector>
 
+void initialize()
+{
+    SelectParams(CBaseChainParams::REGTEST);
+    StartMutedLogging();
```

Comment:
Why can't this be a testing setup with logging disabled, like in the other tests?

Reply:
Unfortunately `TestingSetup` introduces non-determinism which makes coverage vary between runs, see https://github.com/bitcoin/bitcoin/pull/19067#discussion_r429986672 :)

Reply:
What about BasicTestgingSetup then?

Reply:
I think it is the same problem there: I get more setup than I need and a subset of that setup work introduces coverage non-determinism :)

I prefer opt-ing in to exactly the setup I need: it avoids the non-determinism problem and I also think that makes it easier to reason about the dependencies of each fuzzing harness.

Reply:
If non-determinism is an issue for fuzz tests, it is also an issue for unit tests etc. Instead of solving it fresh in every single test case, I'd rather fix it once and for all.

Using hacks that don't clean up after themselves, depend on global state and leave global state dirty is asking for mishaps. Currently the fuzz datadir is set to the default location of the datadir. I don't think I need to repeat why this is bad.

If the basic testing setup is still too bloated, then please add one that is stripped down, so that unit and bench tests can use it as well.

Reply:
I was wrong about `BasicTestingSetup`: it seems like only `TestingSetup` introduces coverage non-determinism.

This shows the issue with `TestingSetup`:

```
# With TestingSetup:
$ src/test/fuzz/kitchen_sink corpus/
…
#20425  NEW    cov: 3867 ft: 12141 corp: 248/49Kb lim: 607 exec/s: 464 rss: 357Mb L: 605/607 MS: 5 InsertRepeatedBytes-CopyPart-InsertByte-ChangeBinInt-CopyPart-
        NEW_FUNC[1/68]: 0x56533ed75c60 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::c_str() const /usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/basic_string.h:2291
        NEW_FUNC[2/68]: 0x56533f03dfc0 in std::function<void ()>::function(std::function<void ()> const&) /usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/std_function.h:655
#20441  NEW    cov: 4384 ft: 12690 corp: 249/49Kb lim: 607 exec/s: 454 rss: 358Mb L: 326/607 MS: 1 CopyPart-
…
#28628        NEW    cov: 4384 ft: 12731 corp: 258/54Kb lim: 652 exec/s: 454 rss: 452Mb L: 652/652 MS: 1 CrossOver-
==6668== libFuzzer: run interrupted; exiting

$ src/test/fuzz/kitchen_sink corpus/
#423    INITED cov: 3867 ft: 12182 corp: 245/48Kb exec/s: 423 rss: 148Mb
```

Notice the non-deterministic coverage increase from 3867 to 4384 which drops back to 3867 when the fuzzer is restarted.

I'll go with `BasicTestingSetup` :)

Reply:
Coverage stability with `BasicTestingSetup` vs `TestingSetup`:

Fuzzer with an empty `test_one_input` function, and an `initialize` function containing `static const BasicTestingSetup basic_testing_setup`:

```
INFO: A corpus is not provided, starting from an empty corpus
#2      INITED cov: 58 ft: 59 corp: 1/1b exec/s: 0 rss: 125Mb
#65536  pulse  cov: 58 ft: 59 corp: 1/1b lim: 652 exec/s: 21845 rss: 145Mb
#131072 pulse  cov: 58 ft: 59 corp: 1/1b lim: 1300 exec/s: 21845 rss: 188Mb
#262144 pulse  cov: 58 ft: 59 corp: 1/1b lim: 2611 exec/s: 21845 rss: 339Mb
#524288 pulse  cov: 58 ft: 59 corp: 1/1b lim: 4096 exec/s: 20164 rss: 715Mb
#1048576        pulse  cov: 58 ft: 59 corp: 1/1b lim: 4096 exec/s: 20560 rss: 716Mb
#2097152        pulse  cov: 58 ft: 59 corp: 1/1b lim: 4096 exec/s: 20560 rss: 716Mb
```

Fuzzer with an empty `test_one_input` function, and an `initialize` function containing `static const TestingSetup testing_setup`:

```
INFO: A corpus is not provided, starting from an empty corpus
#2      INITED cov: 58 ft: 59 corp: 1/1b exec/s: 0 rss: 134Mb
#65536  pulse  cov: 58 ft: 59 corp: 1/1b lim: 652 exec/s: 21845 rss: 155Mb
#131072 pulse  cov: 58 ft: 59 corp: 1/1b lim: 1300 exec/s: 21845 rss: 198Mb
#262144 pulse  cov: 58 ft: 59 corp: 1/1b lim: 2611 exec/s: 21845 rss: 348Mb
#524288 pulse  cov: 58 ft: 59 corp: 1/1b lim: 4096 exec/s: 20164 rss: 715Mb
        NEW_FUNC[1/85]: 0x55e526ad96c0 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string<std::allocator<char> >(char const*, std::allocator<char> const&) /usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/basic_string.h:516
        NEW_FUNC[2/85]: 0x55e526adc5a0 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_local_data() /usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/basic_string.h:180
#910854 NEW    cov: 663 ft: 143 corp: 2/3b lim: 4096 exec/s: 20241 rss: 717Mb L: 2/2 MS: 2 InsertByte-CopyPart-
#1048576        pulse  cov: 663 ft: 143 corp: 2/3b lim: 4096 exec/s: 20560 rss: 717Mb
        NEW_FUNC[1/10]: 0x55e526ad95c0 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::c_str() const /usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/basic_string.h:2291
        NEW_FUNC[2/10]: 0x55e526adc140 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_dispose() /usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/basic_string.h:219
#1830962        NEW    cov: 711 ft: 335 corp: 3/4099b lim: 4096 exec/s: 20572 rss: 717Mb L: 4096/4096 MS: 3 ChangeBinInt-CrossOver-ChangeASCIIInt-
#2097152        pulse  cov: 711 ft: 335 corp: 3/4099b lim: 4096 exec/s: 20360 rss: 717Mb
```